### DISCUSSION TITLE
version 13 head script ld+json

### DISCUSSION THREAD

**Discussion by SkyHyve** [vercel/next.js#43227] [Help] [answered]
I am trying Next.js13 and I want to add a <script type="application/ld+json> to head in head.tsx and am following the guide https://nextjs.org/learn/seo/rendering-and-ranking/metadata however it appears to be for <12.

I am not using the old import Head from 'next/head'; method, just using head.tsx but and continuing using

  <script
    async
    type="application/ld+json"
    dangerouslySetInnerHTML={ldJSON()}
  />

with function ldJSON(){ return { __html: `{ "@context": "https://schema.org/", "@graph": [ {

...

I continue getting error: Expected server HTML to contain a matching <script> in <#document>.

**Comment by icyJoseph** [comment, marked as answer]
This is working fine for me, what next version do you have exactly? And also what's surrounding this code?

```jsx
export default function Head() {
  return (
    <head>
      <script
        type="application/ld+json"
        dangerouslySetInnerHTML={{
          __html: `
            {
                "@context": "https://json-ld.org/contexts/person.jsonld",
                "@id": "http://dbpedia.org/resource/John_Lennon",
                "name": "John Lennon",
                "born": "1940-10-09",
                "spouse": "http://dbpedia.org/resource/Cynthia_Lennon"
              }
            `,
        }}
      ></script>
    </head>
  );
}

```

![Screenshot 2022-11-22 at 10 26 45](https://user-images.githubusercontent.com/21013447/203276843-cf5a9026-50d7-44d1-bae4-fce4e3dfb71b.png)

**Reply by SkyHyve** [reply]
So one file contains the 'base' head elements and that gets exported
```
'use client';

export default function __Head__(params){

  function ldJSON(){
    return {
      __html: `{
        "@context": "https://schema.org",
        "@graph": [
...
      }`,
    };
  }

  return(
    <>
      <link rel="icon" href={`${process.env.BASE_HOST}/favicon.ico`}/>
      <link href={`${process.env.BASE_HOST}/`} rel="canonical"/>
      <link href={`${process.env.BASE_HOST}/`} rel="home"/>
      <base href={`${process.env.BASE_HOST}`}/>

      <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
    ...

      <script
        async
        type="application/ld+json"
        dangerouslySetInnerHTML={ldJSON()}
      ></script>
    </>
  );
}
```

Then another file contains the page specific head components

```
'use client';

import __Head__ from '__common__/__head__';

export default function Head(params){

  return(
    <head>
      <title>Title</title>
      {__Head__(params)}

    ....
    </head>
  );
}
```

**Reply by icyJoseph** [reply]
Mmm if you moved your `ldJSON` function and the script, temporarily, to the head component, does that fix it? I wonder if the `use client` directive has something to say about this...

**Reply by SkyHyve** [reply]
Thanks, managed to get it working, the async was causing problems, only reason I added it in first place was because it wasn't working without the <head> tags, I thought wasn;t necessary because the file is head.tsx

**Reply by icyJoseph** [reply]
Awesome, don't forget to mark as resolved!
